design/fapi-grant-management.md

```diff
@@ -0,0 +1,261 @@
+      {
+         "scope":"contacts read write",
+         "resources":[
+            "https://rs.example.com/api"
+         ]
+      },
+      {
+         "scope":"openid"
+      }
+   ],
+   "claims":[
+      "given_name",
+      "nickname",
+      "email",
+      "email_verified"
+   ],
+   "authorization_details":[
+      {
+         "type":"account_information",
+         "actions":[
+            "list_accounts",
+            "read_balances",
+            "read_transactions"
+         ],
+         "locations":[
+            "https://example.com/accounts"
+         ]
+      }
+   ]
+}
+````
+
+#### API Revoke Grant overview
+To revoke a grant, the client sends a HTTP DELETE request to the grant's resource URL. The authorization server responds with a HTTP status code 204 and an empty response body.
+
+This is illustrated by the following example.
+
+````
+DELETE /grants/TSdqirmAxDa0_-DB_1bASQ 
+Host: as.example.com
+Authorization: Bearer 2YotnFZFEjr1zCsicMWpAA
+
+HTTP/1.1 204 No Content
+````
+The AS MUST revoke the grant and all refresh tokens issued based on that particular grant, it SHOULD revoke all access tokens issued based on that particular grant.
+#### Error Responses:
+
+If the resource URL is unknown, the authorization server responds with HTTP status code 400.
+
+If the client is not authorized to perform a call, the authorization server responds with HTTP status code 403.
+
+If the request lacks a valid access token, the authorization server responds with HTTP status code 401.
+
+
+
+#### Representations of Grant Entity
+
+- **grant_id**
+
+String value identifying an individual grant managed by the AS for a certain client and a certain resource owner. 
+
+- **scopes**
+
+JSON array where every entry contains the scope parameter value and (optionally) any resource parameter value as defined in [@!RFC8707] passed in the same authorization request. The AS MUST maintain the scope and resource values passed in different authorization requests in separate objects of the JSON structure in order to preserve their relationship.
+
+- **claims**
+
+JSON array containing the names of all OpenID Connect claims (see [@!OpenID]) as requested and consented in one or more authorization requests associated with the respective grant.
+
+- **authorization_details**
+
+JSON Object as defined in [@!I-D.ietf-oauth-rar] containing all authorization details as requested and consented in one or more authorization requests associated with the respective grant.
+
+- **status**
+
+- **client_id**
+
+- **created_at**
+
+- **updated_at**
+
+
+
+
+### OAuth Protocol Extensions
+#### Authorization Request
+The spec introduces the authorization request parameters `grant_id` and `grant_management_action`. These parameters can be used with any request serving as authorization request.
+
+- `grant_id`: OPTIONAL. String value identifying an individual grant managed by a particular authorization server for a certain client and a certain resource owner.
```

Comment:
The Grant Management specification stated that `grand_id` is optional. Then does this design document treat the following authorization request patterns?

- `grant_management_action=create` and `grant_id` is specified.
- `grant_management_action=create` and `grant_id` is not specified.
- `grant_management_action=update` and `grant_id` is specified.
- `grant_management_action=update` and `grant_id` is not specified.

AFAIK, the Grant Management API specification does not describe how to do on them. However, it is needed to determine them to write codes even if the specification does not described.

Reply:
I added about how AS will deal with it.

Reply:
I see. Thank you for your clarification.